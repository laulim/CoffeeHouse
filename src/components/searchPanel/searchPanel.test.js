import React from 'react';
import {shallow} from 'enzyme';
import SearchPanel from './searchPanel';

describe('Testing <SearchPanel/>', () => {
  const item = shallow(<SearchPanel/>);

  it('SearchPanel has been rendered', () => {
    expect(item).toMatchSnapshot();
  });

  it('SearchPanel state "term" is empty string', () => {
    expect(item.state().term).toBeString()
  });

});
