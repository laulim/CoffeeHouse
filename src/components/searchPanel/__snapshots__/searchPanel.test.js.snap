// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SearchPanel/> SearchPanel has been rendered 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        action="#"
        className="shop__search"
      >
        <label
          className="shop__search-label"
          htmlFor="filter"
        >
          Looking for
        </label>
        <input
          className="shop__search-input"
          id="filter"
          onChange={[Function]}
          placeholder="start typing here..."
          type="text"
        />
      </form>,
      "className": "col-lg-5 offset-1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "#",
        "children": Array [
          <label
            className="shop__search-label"
            htmlFor="filter"
          >
            Looking for
          </label>,
          <input
            className="shop__search-input"
            id="filter"
            onChange={[Function]}
            placeholder="start typing here..."
            type="text"
          />,
        ],
        "className": "shop__search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Looking for",
            "className": "shop__search-label",
            "htmlFor": "filter",
          },
          "ref": null,
          "rendered": "Looking for",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "shop__search-input",
            "id": "filter",
            "onChange": [Function],
            "placeholder": "start typing here...",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          action="#"
          className="shop__search"
        >
          <label
            className="shop__search-label"
            htmlFor="filter"
          >
            Looking for
          </label>
          <input
            className="shop__search-input"
            id="filter"
            onChange={[Function]}
            placeholder="start typing here..."
            type="text"
          />
        </form>,
        "className": "col-lg-5 offset-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "#",
          "children": Array [
            <label
              className="shop__search-label"
              htmlFor="filter"
            >
              Looking for
            </label>,
            <input
              className="shop__search-input"
              id="filter"
              onChange={[Function]}
              placeholder="start typing here..."
              type="text"
            />,
          ],
          "className": "shop__search",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Looking for",
              "className": "shop__search-label",
              "htmlFor": "filter",
            },
            "ref": null,
            "rendered": "Looking for",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "shop__search-input",
              "id": "filter",
              "onChange": [Function],
              "placeholder": "start typing here...",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing <SearchPanel/> Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        action="#"
        className="shop__search"
      >
        <label
          className="shop__search-label"
          htmlFor="filter"
        >
          Looking for
        </label>
        <input
          className="shop__search-input"
          id="filter"
          onChange={[Function]}
          placeholder="start typing here..."
          type="text"
        />
      </form>,
      "className": "col-lg-5 offset-1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "#",
        "children": Array [
          <label
            className="shop__search-label"
            htmlFor="filter"
          >
            Looking for
          </label>,
          <input
            className="shop__search-input"
            id="filter"
            onChange={[Function]}
            placeholder="start typing here..."
            type="text"
          />,
        ],
        "className": "shop__search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Looking for",
            "className": "shop__search-label",
            "htmlFor": "filter",
          },
          "ref": null,
          "rendered": "Looking for",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "shop__search-input",
            "id": "filter",
            "onChange": [Function],
            "placeholder": "start typing here...",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          action="#"
          className="shop__search"
        >
          <label
            className="shop__search-label"
            htmlFor="filter"
          >
            Looking for
          </label>
          <input
            className="shop__search-input"
            id="filter"
            onChange={[Function]}
            placeholder="start typing here..."
            type="text"
          />
        </form>,
        "className": "col-lg-5 offset-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "#",
          "children": Array [
            <label
              className="shop__search-label"
              htmlFor="filter"
            >
              Looking for
            </label>,
            <input
              className="shop__search-input"
              id="filter"
              onChange={[Function]}
              placeholder="start typing here..."
              type="text"
            />,
          ],
          "className": "shop__search",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Looking for",
              "className": "shop__search-label",
              "htmlFor": "filter",
            },
            "ref": null,
            "rendered": "Looking for",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "shop__search-input",
              "id": "filter",
              "onChange": [Function],
              "placeholder": "start typing here...",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
